cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -22,6 +22,10 @@ use super::recipes::{RecipeGroup, Template};
 use crate::cdsl::formats::FormatRegistry;
 use crate::cdsl::instructions::BindParameter::Any;
 
+enum EncodingPattern {
```

Comment:
How do you feel about naming this CpuMode? (even if there's already something named this way somewhere else, I don't think the two namings would interact or conflict in any way)